File: 2566_Maximum_Difference_by_Remapping_a_Digit.cpp
class Solution {
public:
    // Version 1: Direct string manipulation
    int minMaxDifference(int num) {
        std::string s_num = std::to_string(num);

        // Max Value Calculation
        std::string max_str = s_num;
        char target_max = ' ';
        for (char c : max_str) {
            if (c != '9') {
                target_max = c;
                break;
            }
        }
        if (target_max != ' ') {
            for (char &c : max_str) {
                if (c == target_max) {
                    c = '9';
                }
            }
        }
        int max_val = std::stoi(max_str);

        // Min Value Calculation
        std::string min_str = s_num;
        char target_min = min_str[0];
        for (char &c : min_str) {
            if (c == target_min) {
                c = '0';
            }
        }
        int min_val = std::stoi(min_str);

        return max_val - min_val;
    }
};

File: 2566_Maximum_Difference_by_Remapping_a_Digit.py
class Solution(object):
    def minMaxDifference(self, num):
        """
        :type num: int
        :rtype: int
        """
        s_num = str(num)

        # Calculate max_value
        max_str_list = list(s_num)
        digit_to_change_max = ''
        for digit in max_str_list:
            if digit != '9':
                digit_to_change_max = digit
                break
        
        if digit_to_change_max != '':
            for i in range(len(max_str_list)):
                if max_str_list[i] == digit_to_change_max:
                    max_str_list[i] = '9'
        max_val = int("".join(max_str_list))

        # Calculate min_value
        min_str_list = list(s_num)
        digit_to_change_min = min_str_list[0]
        for i in range(len(min_str_list)):
            if min_str_list[i] == digit_to_change_min:
                min_str_list[i] = '0'
        min_val = int("".join(min_str_list))

        return max_val - min_val
